lib/chain.js

```diff
@@ -157,19 +157,38 @@ function call(handler, err, req, res, _next) {
         _next(nextErr, req, res);
     }
 
+    function resolve() {
+        next();
+    }
+
+    function reject(error) {
+        if (!error) {
+            error = new errors.VError(
+                {
+                    name: 'AsyncHandlerRejection',
```

Comment:
Please define new error types in: https://github.com/restify/node-restify/blob/master/lib/errorTypes.js